From 351e99f06060d53c87278b074a27b67cb6960e6e Mon Sep 17 00:00:00 2001
From: chandrap <prakash1.chandra@intel.com>
Date: Fri, 13 Nov 2015 15:20:27 -0700
Subject: [PATCH] Title: Avalanche Intc and PP Header changes to include
 #ifdef HSD#244296 Issue: HSD:244296 IssueType: Defect
 Symptom: linux/skbuff.h changes Interfere with
 -Werror=undef flag RootCause: used #if rather than #ifdef
 Resolution: change #ifdef flag Impact: Packet Ptrocessor
 related funtionality for datapipe.

---
 include/linux/avalanche/generic/avalanche_pp_api.h |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/linux/avalanche/generic/avalanche_pp_api.h b/include/linux/avalanche/generic/avalanche_pp_api.h
index ebe12fe..f8dc0f6 100755
--- a/include/linux/avalanche/generic/avalanche_pp_api.h
+++ b/include/linux/avalanche/generic/avalanche_pp_api.h
@@ -67,6 +67,14 @@
 
 #ifdef __KERNEL__
 
+#ifdef  CONFIG_ARM_AVALANCHE_PDSP_PP
+ #define PUMA7_SOC_TYPE 1
+ #define PUMA7_OR_NEWER_SOC_TYPE 1
+#else
+ #define PUMA7_SOC_TYPE 0
+ #define PUMA7_OR_NEWER_SOC_TYPE 0
+ #endif
+ #define PUMA6_SOC_TYPE 0
 #if PUMA6_SOC_TYPE
 #include <asm-arm/arch-avalanche/puma6/puma6_cppi_prv.h>
 #endif
-- 
1.7.9.5

